### PR TITLE
Add sg_tests.yml for running SG unit tests

### PR DESCRIPTION
**Pull request by jayahariv**
* Separated out the SG tests to Replicator+SG test suite. 
* Add a new scheme only to run SG unit tests - CBL_ObjC_SG_Tests
* Download and start SG through the GH Action workflow. 
* Run basic tests for SG. 
* Disable the SSL-SG tests.

### CODE REVIEW

File: .github/workflows/sg_tests.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Manual workflow
```
Comment (jayahariv) [inline .github/workflows/sg_tests.yml:1]:
```suggestion
name: Sync gateway uni tests
```

File: .github/workflows/sg_tests.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          echo $device | cat >device
+          echo "Selected the device : ${device}"
+      - name: "Build-for-testing"
+        run: |
+          export CBL_TEST_HOST=localhost
+          DEVICE=$(cat device)
+          xcodebuild build-for-testing -scheme "CBL_ObjC_SG_Tests" -project CouchbaseLite.xcodeproj  -destination "platform=iOS Simulator,name=${DEVICE}"
+      - name: "Test"
+        run: |
+          DEVICE=$(cat device)
```
Comment (jayahariv) [inline .github/workflows/sg_tests.yml:35]:
```suggestion
          export CBL_TEST_HOST=localhost
          DEVICE=$(cat device)
```

File: .github/workflows/sg_tests.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+          tar -zxvf sg.zip
+          ./couchbase-sync-gateway/bin/sync_gateway vendor/couchbase-lite-core/Replicator/tests/data/config.json & sleep 10
+      - name: Select-Device
+        run: |
+          device=`xcrun xctrace list devices 2>&1 | grep -oE 'iPhone.*?[^\(]+' | head -1 | sed 's/Simulator//g' | awk '{$1=$1;print}'`
+          echo $device | cat >device
+          echo "Selected the device : ${device}"
+      - name: "Build-for-testing"
+        run: |
+          export CBL_TEST_HOST=localhost
```
Comment (jayahariv) [inline .github/workflows/sg_tests.yml:30]:
```suggestion
```

File: .github/workflows/sg_tests.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+          DEVICE=$(cat device)
+          xcodebuild build-for-testing -scheme "CBL_ObjC_SG_Tests" -project CouchbaseLite.xcodeproj  -destination "platform=iOS Simulator,name=${DEVICE}"
+      - name: "Test"
+        run: |
+          export CBL_TEST_HOST=localhost
```
Comment (jayahariv) [inline .github/workflows/sg_tests.yml:35]:
```suggestion
```

File: .github/workflows/sg_tests.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+          DEVICE=$(cat device)
+          xcodebuild test-without-building -scheme "CBL_ObjC_SG_Tests" -project CouchbaseLite.xcodeproj  -destination "platform=iOS Simulator,name=${DEVICE}"
```
Comment (jayahariv) [inline .github/workflows/sg_tests.yml:37]:
```suggestion
          xcodebuild test-without-building -scheme "CBL_ObjC_SG_Tests" -project CouchbaseLite.xcodeproj  -destination "platform=iOS Simulator,name=${DEVICE}" CBL_TEST_HOST='localhost'
```

File: .github/workflows/sg_tests.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with: 
+          submodules: recursive
+      - name: Download-and-run-SG
+        run: |
+          curl -o sg.zip https://packages.couchbase.com/releases/couchbase-sync-gateway/3.0.0-beta02/couchbase-sync-gateway-community_3.0.0-beta02_x86_64.zip
+          tar -zxvf sg.zip
+          ./couchbase-sync-gateway/bin/sync_gateway vendor/couchbase-lite-core/Replicator/tests/data/config.json & sleep 10
```
Comment (pasin) [inline .github/workflows/sg_tests.yml:22]: We can run two SG for testing both SSL and non-SSL. There is also a config for SSL in that directory.

Comment (pasin) [reply .github/workflows/sg_tests.yml:22]: Also we should test using SSL instead of non-SSL where possible.

**Pull request closed without merging**